test-data/pointer.yaml

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Prevail Verifier contributors.
+# SPDX-License-Identifier: MIT
+---
+test-case: 32-bit pointer truncation - addition
+
+pre:
+  - "r1.ctx_offset=0"
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.type=ctx"
+  - "r1.uvalue=[0, 4294967295]"
+
+code:
+  <start>: |
+    w1 += 0
+    r2 = *(u32 *)(r1 + 0)
+
+post: []
+
+messages:
+  - "1: Invalid type (r1.type in {ctx, stack, packet, shared})"
+  - "1: Invalid type (valid_access(r1.offset, width=4) for read)"
+
+---
+test-case: 32-bit pointer truncation - subtraction
+
+pre:
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.type=ctx"
+  - "r1.uvalue=[0, 4294967295]"
+
+code:
+  <start>: |
+    w1 -= 0
+    r2 = *(u32 *)(r1 + 0)
+
+post: []
+
+messages:
+  - "1: Invalid type (r1.type in {ctx, stack, packet, shared})"
+  - "1: Invalid type (valid_access(r1.offset, width=4) for read)"
+
+---
+test-case: 32-bit pointer truncation - multiplication
+
+pre:
+  - "r1.ctx_offset=0"
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.type=ctx"
+  - "r1.uvalue=[0, 4294967295]"
+
+code:
+  <start>: |
+    w1 *= 1
+    r2 = *(u32 *)(r1 + 0)
+
+post:
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.uvalue=[0, 4294967295]"
+
+messages:
+  - "0: Invalid type (r1.type == number)"
+  - "1: Invalid type (r1.type in {ctx, stack, packet, shared})"
+  - "1: Invalid type (valid_access(r1.offset, width=4) for read)"
+
+---
+test-case: 32-bit pointer truncation - division
+
+pre:
+  - "r1.ctx_offset=0"
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.type=ctx"
+  - "r1.uvalue=[0, 4294967295]"
+
+code:
+  <start>: |
+    w1 /= 1
+    r2 = *(u32 *)(r1 + 0)
+
+post:
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.uvalue=[0, 4294967295]"
+
+messages:
+  - "0: Invalid type (r1.type == number)"
+  - "1: Invalid type (r1.type in {ctx, stack, packet, shared})"
+  - "1: Invalid type (valid_access(r1.offset, width=4) for read)"
+
+---
+test-case: 32-bit pointer truncation - modulo
+
+pre:
+  - "r1.ctx_offset=0"
+  - "r1.svalue=[0, 4294967295]"
+  - "r1.svalue=r1.uvalue"
+  - "r1.type=ctx"
+  - "r1.uvalue=[0, 4294967295]"
+
+code:
+  <start>: |
+    w1 %= 1
```

Comment:
maybe better to do `%=0` since MOD is defined as
`dst = (src != 0) ? (dst % src) : dst`